Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -191,7 +196,21 @@ protected virtual void Validate()
                             "Emulator environment variable '{0}' is not set, contrary to use of {1}.{2}",
                             variable, nameof(EmulatorDetection), nameof(EmulatorDetection.EmulatorOnly));
                     }
-                    break;
+                    // When the settings *only* support the use of an emulator, the other properties shouldn't be set.
```

Comment:
Yep, I like this better.
One last thing, when EmulatorOnly, should we check that requiredEmulatorEnvironmentVariables has at least one element? I know they are set by client lib code, but it seems almost like a loophole.

Reply:
If we ever had an emulator that didn't need *any* environment variables, setting it to either EmulatorOnly or EmulatorOrProduction should just use it automatically, I guess. It would be pretty weird though.